tests/ssl-opt.sh

```diff
@@ -340,6 +341,23 @@ check_server_hello_time() {
     fi
 }
 
+# terminate server after giving it some time to wrap up and write logs
+terminate_server() {
+    # Shortcut the delay in case we know logs are ready.
+    # This only works for Mbed TLS, and not for all tests (sometimes with DTLS
+    # the server still doesn't know the client has disconnected at the end).
+    # However the shortcut test is designed to only have false negatives and
+    # no false positives, and we fall back to sleeping on negatives.
+    if tail -n-1 $SRV_OUT | grep -F 'Waiting for a remote connection' >/dev/null; then :;
+    else
+        sleep $(( $SRV_DELAY_BASE + $SRV_DELAY_SECONDS ))
```

Comment:
That's a very long timeout just for writing logs.

Reply:
I agree, and part of me was tempted to set `SRV_DELAY_BASE` to `0.1` instead (the other one already is 0 most of the time), but then, we keep seeing deadlines being missed on the CI machines... At this point I guess I just want to drastically reduce the false negative rate on the CI, and only after reaching that goal, work on reducing running time while keeping the false negative rate as close to 0 as possible. So IMO when we need to pick a somewhat arbitrary timeout like here (which hopefully we can get rid of in the future), I'd rather be too conservative than not enough for now.

So unless you strongly disagree, I'm inclined to keep this one to 1 full second.

Reply:
I'm not sure if `sleep` supports fractional arguments on all the platforms we want to support.

Reply:
I might be missing something, but feel like I'm getting contradictory signals here: first you were noticing that 1 second is long, now you seem to be arguing against fractional arguments, and unless you have something else than `sleep` in mind, I don't see how these two remarks can be reconciled into something actionable. Can you clarify if you'd like be to change something here?

Reply:
Oh, sorry, I'm the one who was missing something. I was reading the wrong values and thought that the total was much higher than 1.

But on re-reading this, I think I'm missing something else. Why do this check _before_ sending a signal to the server? If the server is operating normally, then `kill $SRV_PID; wait $SRV_PID` should result in {{ssl_server2}} terminating normally (after running its SIGTERM handler). So shouldn't any timeout be on the `wait`, in case the server is stuck in an infinite loop?

Reply:
I don't think we're trying to address the case where the server is stuck in an infinite loop here: so far we assume it'll always terminate nicely after receiving SIGTERM and I don't think we've noticed the opposite happening so far (fingers crossed).

What this delay was trying to solve is that I suspected that sometimes the server was killed before it had time to write the expected "Last error was ..." message. But now that you comment made me reconsider that, I realised that we shouldn't need any delay for that: we're in control of what the server does when receiving SIGTERM, so it's up to us to ensure that if does print everything it need to print on its way out.

So I think this arbitrary sleep was entirely the wrong way of addressing the problem, and I'm now inclined to withdraw it - see my new take on IOTSSL-2006 on our internal tracker, and thanks for forcing me to clarify my thinking here..